### PR TITLE
chore(e2e-test): [Test automation] Verify Home Page can be customized

### PR DESCRIPTION
**Pull request by teknaS47**
## Description

Added configuration to display the various components that can be added to the Home Page including the default elements.
Added automation to test these Home Page elements to verify that the Home Page can be customized.

Video: 
[video.webm](https://github.com/user-attachments/assets/beaef269-8bb7-4b33-bce1-24ce87d6ea01)

## Which issue(s) does this PR fix

- Fixes [RHIDP-5061](https://issues.redhat.com/browse/RHIDP-5061)

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: .ibm/pipelines/resources/config_map/app-config-rhdh.yaml [hunk 1]
```diff
@@ -17,6 +17,148 @@ app:
 dynamicPlugins:
   rootDirectory: dynamic-plugins-root
   frontend:
+    'janus-idp.backstage-plugin-dynamic-home-page':
```
Comment (christoph-jerolimov) [inline .ibm/pipelines/resources/config_map/app-config-rhdh.yaml:20]: Just one things: After PR #1944 is merged, this should be:

```suggestion
    red-hat-developer-hub.backstage-plugin-dynamic-home-page:
```

Everything else lgtm!

File: e2e-tests/playwright/e2e/home-page-customization.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { test } from "@playwright/test";
+import { UIhelper } from "../utils/ui-helper";
+import { Common } from "../utils/common";
+
+test.describe("Learning Paths", () => {
```
Comment (christoph-jerolimov) [inline e2e-tests/playwright/e2e/home-page-customization.spec.ts:5]: One more small thing:

```suggestion
test.describe("Home page customization", () => {
```

**Review by christoph-jerolimov — APPROVED** [review]
Lgtm!

I didn't checked if they pass!

**Comment by christoph-jerolimov** [comment]
Someone else with QE background should add final lgtm.

/lgtm cancel

File: .ibm/pipelines/resources/config_map/app-config-rhdh.yaml [hunk 3]
```diff
@@ -17,6 +17,148 @@ app:
 dynamicPlugins:
   rootDirectory: dynamic-plugins-root
   frontend:
+    red-hat-developer-hub.backstage-plugin-dynamic-home-page:
```
Comment (subhashkhileri) [inline .ibm/pipelines/resources/config_map/app-config-rhdh.yaml:20]: Can this entire customization configuration be moved to a different location, such as another YAML file? The app-config.yaml file is becoming too large.

Comment (teknaS47) [reply .ibm/pipelines/resources/config_map/app-config-rhdh.yaml]: Created a new config file called dynamic-homepage-and-sidebar-config.yaml

**Comment by christoph-jerolimov** [comment]
@teknaS47 can you please rebase this PR. I guess it will solve the Node.js 20 job issue. Thank you

**Pull request merged**